# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../synaptics/compile-fix.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
diff -Nur synaptics-0.14.3.orig/Xincludes/usr/X11R6/include/X11/Xfuncproto.h synaptics-0.14.3/Xincludes/usr/X11R6/include/X11/Xfuncproto.h
--- synaptics-0.14.3.orig/Xincludes/usr/X11R6/include/X11/Xfuncproto.h	2005-07-02 13:50:40.000000000 +0300
+++ synaptics-0.14.3/Xincludes/usr/X11R6/include/X11/Xfuncproto.h	2005-09-25 21:14:39.000000000 +0300
@@ -69,4 +69,12 @@
 #endif
 #endif /* _XFUNCPROTOBEGIN */
 
+#if defined(__GNUC__) && (__GNUC__ >= 4) 
+# define _X_SENTINEL(x) __attribute__ ((__sentinel__(x))) 
+# define _X_ATTRIBUTE_PRINTF(x,y) __attribute__((__format__(__printf__,x,y))) 
+#else 
+# define _X_SENTINEL(x) 
+# define _X_ATTRIBUTE_PRINTF(x,y) 
+#endif /* GNUC >= 4 */ 
+
 #endif /* _XFUNCPROTO_H_ */
